src/accelerate/utils/offload.py

```diff
@@ -152,12 +153,25 @@ def __init__(
         self.index = {} if index is None else index
         self.all_keys = list(self.state_dict.keys())
         self.all_keys.extend([key for key in self.index if key not in self.all_keys])
+        self.device = device
 
     def __getitem__(self, key: str):
         # State dict gets priority
         if key in self.state_dict:
             return self.state_dict[key]
         weight_info = self.index[key]
+        if weight_info.get("safetensors_file") is not None:
+            from safetensors import safe_open
```

Comment:
We need a guard here for `is_safetensors_available` and raise an error if not :)

Reply:
Indeed, will clean that up on Monday!